src/main/java/com/atlassian/bitbucket/jenkins/internal/link/BitbucketJobLinkActionFactory.java

```diff
@@ -55,25 +56,21 @@ public BitbucketJobLinkActionFactory(BitbucketExternalLinkUtils externalLinkUtil
                             .orElse(Collections.emptyList());
                 }
             }
-            // Multibranch Pipeline Job built with an SCMStep
-            if (getWorkflowSCMs(workflowJob).stream().anyMatch(scm -> scm instanceof BitbucketSCM)) {
-                Optional<BitbucketSCMRepository> maybeRepository = getWorkflowSCMs(workflowJob)
-                        .stream()
-                        .filter(scm -> scm instanceof BitbucketSCM)
-                        .map(scm -> ((BitbucketSCM) scm).getBitbucketSCMRepository())
-                        .findFirst();
-                return maybeRepository.flatMap(scmRepository -> externalLinkUtils.createBranchDiffLink(scmRepository, target.getName()))
+            // Multibranch Pipeline Job
+            if (getWorkflowParent(workflowJob) instanceof WorkflowMultiBranchProject) {
+                // Multibranch Pipeline Job from SCM Source, or if there is none, try and get it from an SCMStep
+                return Stream.of(getScmSource(workflowJob), getScmStep(workflowJob))
```

Comment:
Maybe a small inline comment here to remark on the significance of the ordering would be good